.circleci/config.yml

```diff
@@ -1,89 +1,189 @@
-version: 2.1
+#
+# Copyright 2023 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+#
 
+version: 2.1
 orbs:
-  android: circleci/android@2.0
+  # Use the circleci android orb version that supports OpenJDK17 required by AGP 8.2+
+  android: circleci/android@2.4.0
+  codecov: codecov/codecov@4.0.1
+
+parameters:
+  gradle-cache-prefix:
+    type: string
+    default: v1
+  build-cache-prefix:
+    type: string
+    default: v1
+  build-path:
+    type: string
+    default: "code/media/build"
+
+# Workflows orchestrate a set of jobs to be run;
+workflows:
+  version: 2
+  build-test-deploy:
+    jobs:
+      - validate-code
+      - build-and-unit-test:
+          requires:
+            - validate-code
+      - functional-test:
+          requires:
+            - validate-code
+      - build-test-app:
+          requires:
+            - validate-code
+          filters:
+            branches:
+              only:
+                - main
+                - staging
 
 jobs:
-  build-and-unit-test:
+  validate-code:
     executor:
-      name: android/android-machine
-      resource-class: large
-      tag: 2022.01.1
+      name: android/android-docker
+      resource-class: medium
+      tag: 2024.01.1
 
     steps:
       - checkout
+
+      - android/restore-gradle-cache:
+          cache-prefix: << pipeline.parameters.gradle-cache-prefix >>
+
+      - android/restore-build-cache:
+          cache-prefix: << pipeline.parameters.build-cache-prefix >>
+
       - run:
-          name: Check format
+          name: Check Format
           command: make checkformat
 
       - run:
-          name: Check style
+          name: Check Style
           command: make checkstyle
 
+      - android/save-gradle-cache:
+          cache-prefix: << pipeline.parameters.gradle-cache-prefix >>
+
+      - android/save-build-cache:
+          cache-prefix: << pipeline.parameters.build-cache-prefix >>
+
+      - store_artifacts:
+          path: code/<< pipeline.parameters.build-path >>/build/reports  
```

Comment:
path: code/<< pipeline.parameters.build-path >>/reports


We had a extra /build